#!/usr/bin/env python
PACKAGE="sim_clock"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# TODO(lucasw) negative time is an interesting possibility
# but almost certainly works badly with other time consuming nodes.
gen.add("dt", double_t, 1, "time delta", 0.005, 0.001, 0.1)
gen.add("real_time_factor", double_t, 1, "real time factor", 1.0, 0.001, 10.0)
gen.add("play_pause", bool_t, 1, "play/pause", True)
gen.add("single_step", bool_t, 1, "single step by step time amount if paused", False)
gen.add("step_time", double_t, 1, "amount to single step", 0.02, 0.001, 1.0)
gen.add("single_dt", bool_t, 1, "single step by dt amount paused", False)

exit(gen.generate(PACKAGE, "sim_clock", "Clock"))
